odoo/addons/base/models/ir_default.py

```diff
@@ -188,3 +207,103 @@ def discard_values(self, model_name, field_name, values):
         json_vals = [json.dumps(value, ensure_ascii=False) for value in values]
         domain = [('field_id', '=', field.id), ('json_value', 'in', json_vals)]
         return self.search(domain).unlink()
+
+    def _evaluate_leaf_with_fallback(self, model_name, leaf):
+        """
+        when the field value of the leaf is company_dependent without
+        customization, evaluate if its fallback value will be filtered out by
+        the leaf
+        return True/False/None(for unknown)
+        """
+        (key, comparator, value) = leaf
+        field_name, *rest = key.split('.', 1)
+        model = self.env[model_name]
+        fallback = model._fields[field_name].get_company_dependent_fallback(model)
+        if isinstance(fallback, models.BaseModel):
+            # TODO cwg: TBD maybe directly return None
+            #  when comparator in ['company_id', 'company_ids'] and mapped result is not res.company
+            # which will cause search for non res.company records
+            if rest:
+                return bool(fallback.filtered_domain([(rest[0], comparator, value)]))
+            if comparator in ['child_of', 'parent_of']:
+                return bool(fallback.filtered_domain([('id', comparator, value)]))
+            data = fallback
+        else:
+            data = [fallback]
+
+        field = self.env[model_name]._fields[field_name]
```

Comment:
The code below seems like a copy paste from `filtered_domain`. Could you refactor this?